flink-external-resources/flink-external-resource-gpu/src/main/java/org/apache/flink/externalresource/gpu/GPUDriver.java

```diff
@@ -0,0 +1,134 @@
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.externalresource.gpu;
+
+import org.apache.flink.annotation.VisibleForTesting;
+import org.apache.flink.api.common.externalresource.ExternalResourceDriver;
+import org.apache.flink.configuration.ConfigConstants;
+import org.apache.flink.configuration.ConfigOption;
+import org.apache.flink.configuration.Configuration;
+import org.apache.flink.configuration.IllegalConfigurationException;
+import org.apache.flink.util.FlinkException;
+import org.apache.flink.util.Preconditions;
+import org.apache.flink.util.StringUtils;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.InputStreamReader;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+import static org.apache.flink.configuration.ConfigOptions.key;
+
+/**
+ * Driver takes the responsibility to discover GPU resources and provide the GPU resource information.
+ * It get the GPU information by executing user-defined discovery script.
+ */
+class GPUDriver implements ExternalResourceDriver {
+
+	private static final Logger LOG = LoggerFactory.getLogger(GPUDriver.class);
+
+	@VisibleForTesting
+	static final ConfigOption<String> DISCOVERY_SCRIPT_PATH =
+		key("discovery-script.path")
+			.stringType()
+			.noDefaultValue();
+
+	@VisibleForTesting
+	static final ConfigOption<String> DISCOVERY_SCRIPT_ARG =
+		key("discovery-script.args")
+			.stringType()
+			.noDefaultValue();
+
+	private final File discoveryScript;
+	private final String args;
+
+	GPUDriver(Configuration config) throws Exception {
+		final String discoveryScriptPath = config.getString(DISCOVERY_SCRIPT_PATH);
+		if (StringUtils.isNullOrWhitespaceOnly(discoveryScriptPath)) {
+			throw new IllegalConfigurationException(
+				"Could not find config of the path of gpu discovery script, the relevant config key is {}.", "external-resource.<resource-name>." + DISCOVERY_SCRIPT_ARG.key());
+		}
+
+		discoveryScript = new File(System.getenv().getOrDefault(ConfigConstants.ENV_FLINK_HOME_DIR, ".") +
+			"/" + discoveryScriptPath);
```

Comment:
I think we should consider supporting both absolute and relative path for this configuration option.